tests/test_discriminated_union.py

```diff
@@ -940,43 +935,62 @@ def test_lax_or_strict_definitions() -> None:
+    # insert_assert(discriminated_schema)
     assert discriminated_schema == {
-        'definitions': [{'ref': 'my-str-definition', 'type': 'str'}],
+        'type': 'definitions',
         'schema': {
+            'type': 'tagged-union',
             'choices': {
+                'cat': {
+                    'type': 'typed-dict',
+                    'fields': {
+                        'kind': {'type': 'typed-dict-field', 'schema': {'type': 'literal', 'expected': ['cat']}}
+                    },
+                },
                 'DOG': {
+                    'type': 'lax-or-strict',
                     'lax_schema': {
+                        'type': 'typed-dict',
                         'fields': {
-                            'kind': {'schema': {'expected': ['DOG'], 'type': 'literal'}, 'type': 'typed-dict-field'}
+                            'kind': {'type': 'typed-dict-field', 'schema': {'type': 'literal', 'expected': ['DOG']}}
                         },
-                        'type': 'typed-dict',
                     },
                     'strict_schema': {
-                        'definitions': [{'ref': 'my-int-definition', 'type': 'int'}],
+                        'type': 'definitions',
                         'schema': {
+                            'type': 'typed-dict',
                             'fields': {
-                                'kind': {'schema': {'expected': ['dog'], 'type': 'literal'}, 'type': 'typed-dict-field'}
+                                'kind': {'type': 'typed-dict-field', 'schema': {'type': 'literal', 'expected': ['dog']}}
                             },
-                            'type': 'typed-dict',
                         },
-                        'type': 'definitions',
+                        'definitions': [{'type': 'int', 'ref': 'my-int-definition'}],
                     },
-                    'type': 'lax-or-strict',
                 },
-                'cat': {
-                    'fields': {
-                        'kind': {'schema': {'expected': ['cat'], 'type': 'literal'}, 'type': 'typed-dict-field'}
+                'dog': {
+                    'type': 'lax-or-strict',
+                    'lax_schema': {
+                        'type': 'typed-dict',
+                        'fields': {
+                            'kind': {'type': 'typed-dict-field', 'schema': {'type': 'literal', 'expected': ['DOG']}}
+                        },
+                    },
+                    'strict_schema': {
+                        'type': 'definitions',
+                        'schema': {
+                            'type': 'typed-dict',
+                            'fields': {
+                                'kind': {'type': 'typed-dict-field', 'schema': {'type': 'literal', 'expected': ['dog']}}
+                            },
+                        },
+                        'definitions': [{'type': 'int', 'ref': 'my-int-definition'}],
                     },
```

Comment:
I don't know what this means

Reply:
We used to do recursive lookups. If the *value* of `choices` was an int or a str (instead of a CoreSchema) we'd look up that key. So given `{'DOG': {'type': 'int'}, 'dog': 'DOG', 'pet': 'dog'}` we would build validators like `{'DOG': {'type': 'int'}, 'dog': {'type': 'int'}, 'pet': {'type': 'int'}}`. I'm getting rid of that here in favor of having that happen in pydantic / whatever is building the schema.